sklearn/metrics/tests/test_classification.py

```diff
@@ -1383,6 +1383,24 @@ def test_average_binary_jaccard_score(recwarn):
     assert not list(recwarn)
 
 
+def test_zero_devision_jaccard_score():
+    y_true = np.array([[1, 0, 1], [0, 0, 0]])
+    y_pred = np.array([[0, 0, 0], [0, 0, 0]])
+    msg = ('Jaccard is ill-defined and being set to 0.0 in '
+           'samples with no true or predicted labels.'
+           ' Use `zero_division` parameter to control this behavior.')
+    assert assert_warns_message(UndefinedMetricWarning, msg,
+                                jaccard_score,
+                                y_true,
+                                y_pred,
+                                average='samples',
+                                zero_division='warn') == 0.0
```

Comment:
We have a preference for `pytest.warns`:

```py
with pytest.warns(UndefinedMetricWarning, match=msg):
    score = jaccard_score(y_true, y_pred, ...)
assert score == pytest.approx(score)
```

Reply:
Resolved by a083dcf99e1fc4250064d6d6c549c652565f878c